rs_client/staging_client.py

```diff
@@ -0,0 +1,265 @@
+            job_id (str): identifier of the current job
+        """
+        stac_input_dict = {}
+        # If stac_input is a file, load this file to a dictionary
+        if isinstance(stac_input, str):
+            # If the input is a valid path to a json_file, load this file
+            if os.path.exists(os.path.dirname(stac_input)) and stac_input.endswith(".json"):
+                # Read the yaml or json file
+                with open(stac_input, encoding="utf-8") as opened:
+                    stac_file_to_dict = json.loads(opened.read())
+                    stac_input_dict = stac_file_to_dict
+            # If the input string is not a path, try to convert the content of the string to a json dictionary
+            else:
+                stac_input_dict = json.loads(stac_input)
+        else:
+            stac_input_dict = stac_input
+
+        # Check that the type
+        if "type" not in stac_input_dict:
+            raise KeyError("Staging input data has missing key 'type'")
+
+        # Validate input data using Pydantic
+        if stac_input_dict["type"] == "Feature":
+            stac_item = Item(**stac_input_dict)
+            stac_item_collection = ItemCollection(
+                **{
+                    "type": "FeatureCollection",
+                    "context": {"limit": 1000, "returned": 2},
+                    "features": [stac_item],
+                },  # type: ignore
+            )
+        else:
+            stac_item_collection = ItemCollection(**stac_input_dict)
+
+        staging_body = {  # pylint: disable=line-too-long
+            "version": "0.2.0",
+            "id": "staging",
+            "title": {"en": "Staging"},
+            "description": {
+                "en": "A process that takes an external STAC ItemCollection, asynchronously download"
+                "its assets into the RS catalog bucket and creates the corresponding STAC items in the RS catalog.",
+            },
+            "jobControlOptions": ["async-execute"],
+            "keywords": ["stac", "staging"],
+            "links": [
+                {
+                    "type": "text/html",
+                    "rel": "about",
+                    "title": "documentation",
+                    "href": "https://home.rs-python.eu/rs-documentation/rs-server"
+                    "/docs/doc/users/functionalities/#staging",
+                    "hreflang": "en-US",
+                },
+            ],
+            "inputs": {
+                "collection": {
+                    "title": "Target collection",
+                    "description": "The target collection identifier in the RS catalog",
+                    "id": out_coll_name,
+                    "schema": {"type": "string"},
+                    "minOccurs": 1,
+                    "maxOccurs": 1,
+                },
+                "items": stac_item_collection.model_dump(mode="json"),
+                "provider": "cadip",
+            },
+            "outputs": {
```

Comment:
This need to be discussed, at some point output should be configurable.

Reply:
@vprivat-ads can we have a discussion about this ?